qrexec/policy/parser.py

```diff
@@ -590,14 +611,13 @@ class DispVMTag(Source, Target):
 
     def match(
         self,
-        other: Optional[str],
+        other: str,
         *,
         system_info: FullSystemInfo,
         source: Optional[VMToken]=None
     ) -> bool:
-        if isinstance(other, DispVM):
-            assert source is not None
-            other = other.get_dispvm_template(source, system_info=system_info)
+        if isinstance(other, DispVM) and source is not None:
```

Comment:
I'd prefer the assert to stay, instead of silently skipping this branch in that case. Or is None as source valid in this case now for some reason?